internal/ingress/controller/parser/kongstate/route.go

```diff
@@ -303,4 +305,54 @@ func (r *Route) overrideByKongIngress(log logrus.FieldLogger, kongIngress *confi
 		}
 		r.SNIs = SNIs
 	}
+	if ir.RequestBuffering != nil {
+		r.RequestBuffering = kong.Bool(*ir.RequestBuffering)
+	}
+	if ir.ResponseBuffering != nil {
+		r.ResponseBuffering = kong.Bool(*ir.ResponseBuffering)
+	}
+}
+
+// overrideRequestBuffering ensures defaults for the request_buffering option
+func (r *Route) overrideRequestBuffering(log logrus.FieldLogger, anns map[string]string) {
+	// by default request_buffering should be enabled. It only needs to be
+	// disabled under special use cases, including large payloads.
+	r.RequestBuffering = kong.Bool(true)
```

Comment:
This should remain unset/should not have a default; we should only set the annotation value if it is present. Not setting required fields with default values is fine: Kong will set it on the resulting entity automatically if it's omitted from the admin API request. We generally generally want to avoid duplicating Kong's behavior when possible, and for this setting, need to omit it by default to maintain compatibility with versions older than 2.2.

There are some exceptions where we do set defaults (`strip_path` and `preserve_host`) to conform to K8S Ingress spec requirements, but otherwise we omit them

Reply:
This should also address the issue with existing tests breaking because they don't expect those settings in the target object.

Reply:
Sorry, I had already realized this and had been changing this before you posted this!